src/hooks/useSearchParamState.ts

```diff
@@ -138,11 +159,23 @@ const useSearchParamsState = ({
       // Find any key/value pairs that are in initial, but not in searchParams
       Object.entries(initial).filter(([key]) => !searchParams.has(key))
     );
-    if (Object.keys(toUpdate).length) {
-      // Spread currentParams to avoid overwriting search params set by other components
-      const currentParams = getAllCurrentParams(searchParams);
-      setSearchParams({ ...currentParams, ...toUpdate }, { replace: true });
-    }
+
+    const accumulator = new URLSearchParams();
+    populateParams(toUpdate, accumulator);
+    if (accumulator.entries.length === 0) return; // If we don't have anything to update, return early
```

Comment:
`accumulator.entries` [is a function](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/entries) so I think this is always returning zero and not doing anything. I'm not sure how to even reproduce a bug here though.

If the intention is to return early when there are no params to update, can we instead base that on `Object.keys(toUpdate).length` up on line 162? Or is the population logic relevant here? (I had a question on that method logic too)